src/pure_twopset.erl

```diff
@@ -0,0 +1,125 @@
+%%
+%% Copyright (c) 2015-2016 Christopher Meiklejohn.  All Rights Reserved.
+%% Copyright (c) 2007-2012 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+%% @doc Pure twoPSet CRDT: pure op-based two-phase set.
+%%
+%% @reference Carlos Baquero, Paulo Sérgio Almeida, and Ali Shoker
+%%      Making Operation-based CRDTs Operation-based (2014)
+%%      [http://haslab.uminho.pt/ashoker/files/opbaseddais14.pdf]
+
+-module(pure_twopset).
+-author("Georges Younes <georges.r.younes@gmail.com>").
+
+-behaviour(type).
+
+-define(TYPE, ?MODULE).
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+-export([new/0, new/1]).
+-export([mutate/3, query/1, equal/2]).
+
+-export_type([pure_twopset/0, pure_twopset_op/0]).
+
+-opaque pure_twopset() :: {?TYPE, payload()}.
+-type payload() :: {pure_type:polog(), {ordsets:set(), ordsets:set()}}.
+-type pure_twopset_op() :: {add, pure_type:element()} | {rmv, pure_type:element()}.
+
+%% @doc Create a new, empty `pure_twopset()'
+-spec new() -> pure_twopset().
+new() ->
+    {?TYPE, {orddict:new(), {ordsets:new(), ordsets:new()}}}.
+
+%% @doc Create a new, empty `pure_twopset()'
+-spec new([term()]) -> pure_twopset().
+new([]) ->
+    new().
+
+%% @doc Update a `pure_twopset()'.
+-spec mutate(pure_twopset_op(), pure_type:id(), pure_twopset()) ->
+    {ok, pure_twopset()}.
+mutate({add, Elem}, _VV, {?TYPE, {POLog, {Pure2PAddSet, Pure2PRmvSet}}}) ->
+    AlreadyRemoved = ordsets:is_element(Elem, Pure2PRmvSet),
+    case AlreadyRemoved of
+        true ->
+            {ok, {?TYPE, {POLog, Pure2PAddSet, Pure2PRmvSet}}};
+        false ->
+            PureTwoPSet = {?TYPE, {POLog, {ordsets:add_element(Elem, Pure2PAddSet), Pure2PRmvSet}}},
+            {ok, PureTwoPSet}
+    end;
+mutate({rmv, Elem}, _VV, {?TYPE, {POLog, {Pure2PAddSet, Pure2PRmvSet}}}) ->
+    PureTwoPSet = {?TYPE, {POLog, {ordsets:del_element(Elem, Pure2PAddSet), ordsets:add_element(Elem, Pure2PRmvSet)}}},
```

Comment:
Is safe. Next example is okay.
```erlang
ordsets:del_element(2, ordsets:new()).
```

This is also the semantics I implemented [state_twopset](https://github.com/lasp-lang/types/blob/master/src/state_twopset.erl#L87-L101).

Should we have both?